### ISSUE TITLE
appsettings.json nlog connectionstring does not work with dotnet user-secrets

### ISSUE THREAD

**Issue by paulb2015** [JDetmar/NLog.Extensions.AzureStorage#125]
If the connection string is configured via dotnet user-secrets, nlog logs the following error in its internal log and is unable to log to blob storage.

Error AzureBlobStorageTarget(Name=blob): Failed to create BlobClient with connectionString=. Exception: System.ArgumentNullException: Value cannot be null. (Parameter 'connectionString')

Using appsettings.json (snipped):

`"connectionString": "${configsetting:ConnectionStrings.Storage}"`
``
"ConnectionStrings": {
        "Storage": ""
}
``

See https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets

**Comment by snakefoot** [comment]
Yes right now the BlobStorage-Target requires that the ConnectionString (or ServiceUrl) is available when initializing the NLog Configuration.

This means user-secrets must be loaded upfront, before loading NLog Configuration. 

Maybe it is easier to use ServiceUrl with Managed Client Identity ?

**Comment by fuzzzerd** [comment]
I just discovered this issue myself; and I'm wondering is it possible that the same issue would exist for using the Azure App Service Configuration section from the portal?

**Comment by snakefoot** [comment]
There is a convience method available when using NLog.Web.AspNetCore:
```c#
// Loads appsetting.json and enables ${configsetting}
var logger = LogManager.Setup()
                       .LoadConfigurationFromAppSettings()
                       .GetCurrentClassLogger();
```
One can also do it manually:
```c#

```
See also: https://github.com/NLog/NLog/wiki/ConfigSetting-Layout-Renderer

See also: https://github.com/NLog/NLog.Extensions.Logging/wiki/NLog-configuration-with-appsettings.json
